processor/spanmetricsprocessor/processor.go

```diff
@@ -262,8 +270,15 @@ func (p *processorImp) collectLatencyMetrics(ilm pdata.InstrumentationLibraryMet
 		dpLatency.SetSum(p.latencySum[key])
 
 		setLatencyExemplars(p.latencyExemplarsData[key], timestamp, dpLatency.Exemplars())
-
-		p.metricKeyToDimensions[key].CopyTo(dpLatency.Attributes())
+		if item, ok := p.metricKeyToDimensions.Get(key); ok {
```

Comment:
Can we avoid this duplication with line 300? Also, can we add a comment here briefly describing what this block does, since the logic is more complex now.